stdlib/public/Darwin/Foundation/String.swift

```diff
@@ -54,10 +73,47 @@ extension String : _ObjectiveCBridgeable {
   public static func _unconditionallyBridgeFromObjectiveC(
     _ source: NSString?
   ) -> String {
+    if let source = source {
+      #if !(arch(i386) || arch(arm))
+      if let result = _bridgeTaggedCocoaString(source) {
+        return result
+      }
+      #endif
+      
+      let sourceClass:AnyClass = _getClass(source)
+      
+      if _isRebridgedSwiftStringClass(sourceClass) {
+        return _rebridgeSwiftString(source, sourceClass)
+      }
+      
+      let len = _length(source)
+
+      //We *may* not be able to form a SmallString from this, but above 15 we
+      //definitely can't. It's worth trying because eager bridging is almost
+      //always good if we don't have to allocate for it.
+      if len <= 15 && sourceClass == nscfClass,
```

Comment:
Not hardcoding 15 here would be good… if I can work out how. _SmallString.capacity isn't public as far as I know.

Reply:
We can make it public. How much of this code needs to live in the overlay? Also, it would be 10 on 32-bit.

Reply:
Where this code lives has different tradeoffs: in the overlay, we can do libobjc and NSString calls without having to through shims. In the stdlib, we can use Builtins and _SmallString/_StringGuts members.

The main thing I want to avoid is unnecessary hopping back and forth between them, since that's across a resilience boundary.

Reply:
And balanced against the number of new public decls we’re publishing.